src/olympia/search/tests/test_views.py

```diff
@@ -622,21 +622,75 @@ def test_search_doesnt_return_unlisted_addons(self):
         assert addon.pk not in self.get_results(response)
 
     def test_webextension_boost(self):
-        web_extension = self.addons[1]
+        web_extension = list(sorted(
+            self.addons, key=lambda x: x.name, reverse=True))[0]
         web_extension.current_version.files.update(is_webextension=True)
         web_extension.save()
         self.refresh()
 
-        response = self.client.get(self.url, {'q': 'addon'})
+        response = self.client.get(self.url, {'q': 'Addon'})
         result = self.get_results(response, sort=False)
         assert result[0] != web_extension.pk
 
         create_switch('boost-webextensions-in-search')
         # The boost chosen should have made that addon the first one.
-        response = self.client.get(self.url, {'q': 'addon'})
+        response = self.client.get(self.url, {'q': 'Addon'})
         result = self.get_results(response, sort=False)
         assert result[0] == web_extension.pk
 
+    def test_score_boost_name_match(self):
+        addons = [
```

Comment:
I think that this test and the test below work in isolation, and work by chance in the full run for now, but may cause future tests to break. This is because `ESTestCase` does not unindex add-ons created during the test (we have `empty_index()` for that, but this class does not call it automatically).

Other tests in that class work around this by only modifying add-ons that have been created by the base class, which is `ESTestCaseWithAddons`, which does explicitly unindex add-ons it created (and only those)

Stray add-ons in the index is not always noticeable in other tests because of filtering, but they generally affect scoring, sometimes in subtle ways, making tests around name matching more brittle.